src/Model.cpp

```diff
@@ -59,10 +59,12 @@ void Model::init() {
 
 void Model::save(const std::string &ckpt) {
 #ifdef TENSORFLOW_C_TF_TSTRING_H_
-    std::unique_ptr<TF_TString, decltype(&TF_TString_Dealloc)> tstr(new TF_TString, &TF_TString_Dealloc);
-    TF_TString_Copy(tstr.get(), ckpt.c_str(), ckpt.size());
+    TF_TString tstr;
+    TF_TString_Init(&tstr);
+    TF_TString_Copy(&tstr, ckpt.c_str(), ckpt.size());
+    std::unique_ptr<TF_TString, decltype(&TF_TString_Dealloc)> up_tstr(&tstr, &TF_TString_Dealloc);
```

Comment:
Do we need this? We do not seem to be using this pointer at all.
Also, since tstr is on stack, what would it even mean to release that memory once it goes out of scope? I am not completely sure about this.

Reply:
`TFtring_Dealloc` has to be called to de-allocate the string, so we either call it manually or via the `unique_ptr`. Plus, since `TF_TString_Init` and `TF_TString_Copy` take a `*TF_TString`, they can dynamically (re)allocate memory for the passed in variable, no? Plus, I'd expect ASan to complain if this was fishy.

I'll try ASan without the `unique_ptr`.

Reply:
If I remove the `unique_ptr` line and run the ASan test I get the following failure:
```
=================================================================
==73==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 48 byte(s) in 1 object(s) allocated from:
    #0 0x49e3fd in malloc (/home/docker/kiravision/test/asan-memtest+0x49e3fd)
    #1 0x50df22 in TF_TString_ResizeUninitialized(TF_TString*, unsigned long) /opt/tensorflow/include/tensorflow/core/platform/ctstring_internal.h:276:23
    #2 0x50b9e4 in TF_TString_Copy(TF_TString*, char const*, unsigned long) /opt/tensorflow/include/tensorflow/core/platform/ctstring_internal.h:389:17
    #3 0x50bb59 in Model::restore(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) /home/docker/kiravision/cppflow/src/Model.cpp:110:5
    #4 0x4e9882 in KiraVision::LearnedModel::LearnedModel(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) /home/docker/kiravision/src/LearnedModel.cpp:12:13
    #5 0x4e28bd in KiraVision::TableDetector::TableDetector(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, KiraVision::LogLevel const&) /home/docker/kiravision/src/TableDetector.cpp:13:24
    #6 0x4d057a in main /home/docker/kiravision/test/memtest.cpp:16:31
    #7 0x7ff4ce7100b2 in __libc_start_main /build/glibc-eX1tMB/glibc-2.31/csu/../csu/libc-start.c:308:16

SUMMARY: AddressSanitizer: 48 byte(s) leaked in 1 allocation(s).
make: *** [Makefile:183: run-asan-memtest] Error 1
```

So I think we do need the `unique_ptr`, or to call `TF_TSring_Dealloc` manually.